presto-main/src/main/java/com/facebook/presto/metadata/MetadataManager.java

```diff
@@ -970,6 +970,24 @@ public void revokeTablePrivileges(Session session, QualifiedObjectName tableName
         return ImmutableList.copyOf(grantInfos.build());
     }
 
+    @Override
+    public boolean isWritingStagingFilesEnabled(Session session)
+    {
+        CatalogMetadata catalogMetadata = getCatalogMetadataForWrite(session, session.getCatalog().orElseThrow(() -> new IllegalStateException("No catalog specified")));
+        ConnectorId connectorId = catalogMetadata.getConnectorId();
+        ConnectorMetadata metadata = getMetadata(session, connectorId);
+        return metadata.isWritingStagingFilesEnabled(session.toConnectorSession(connectorId));
+    }
+
+    @Override
+    public void commitStagingFiles(Session session, SchemaTableName tableName, Collection<Slice> fragments)
+    {
+        CatalogMetadata catalogMetadata = getCatalogMetadataForWrite(session, session.getCatalog().orElseThrow(() -> new IllegalStateException("No catalog specified")));
```

Comment:
Ditto about catalog name + the catalog name from the `tableName` should be preferred over session.

Reply:
I removed the SPI changes so this is inlined inside `HiveMetadata`. I guess we don't need catalog there, but I'll leave it unresolved for now so you are aware.